frontend/src/router/RouteGuards.ts

```diff
@@ -1,61 +1,71 @@
-import { Auth } from "aws-amplify";
+import { authState } from "@/providers/authState";
 import type { RouteLocationNormalized, NavigationGuardNext } from "vue-router";
 
 /**
- * Check if the user is authenticated using AWS Amplify
- * @returns {Promise<boolean>}
+ * Waits for the authentication state to be restored.
+ * @returns {Promise<void>}
  */
-export async function isAuthenticated(): Promise<boolean> {
-    const startTime = performance.now(); // Start time
-    try {
-        await Auth.currentSession();
-        const endTime = performance.now(); // End time
-        console.log(
-            `isAuthenticated Execution Time: ${
-                endTime - startTime
-            } milliseconds`
-        );
-        return true;
-    } catch {
-        const endTime = performance.now(); // End time
-        console.log(
-            `isAuthenticated Execution Time: ${
-                endTime - startTime
-            } milliseconds`
-        );
-        return false;
-    }
-}
+const waitForAuthRestoration = (): Promise<void> => {
+    return new Promise((resolve) => {
+        const checkAuthRestored = () => {
+            if (authState.value.isAuthRestored) {
+                resolve();
+            } else {
+                setTimeout(checkAuthRestored, 100);
+            }
+        };
+        checkAuthRestored();
+    });
+};
 
 /**
- * Auth guard to manage navigation based on authentication state.
+ * Checks if the user is authenticated based on the current authentication state.
+ * @returns {boolean} True if the user is authenticated, otherwise false.
+ */
+const isAuthenticated = (): boolean => {
+    return authState.value.isAuthenticated;
+};
+
+/**
+ * Checks if the user is authenticated using AWS Amplify.
+ * Logs the execution time of the check.
+ * @returns {Promise<boolean>} True if the user is authenticated, otherwise false.
+ */
+export const checkAuthentication = async (): Promise<boolean> => {
+    return isAuthenticated();
+};
```

Comment:
Wondering if the "Logs the execution time of the check" is missing? The time does not seems to be logged.
Probably also there is no need for using "async" since no need to "await" the `isAuthenticated` function result.

Reply:
I removed it because it was only for me to assess the performance, with this new change the code execution time will be 0ms 99% of the time